hardware_interface/test/test_components/test_system.cpp

```diff
@@ -78,9 +131,22 @@ class TestSystem : public SystemInterface
     return command_interfaces;
   }
 
-  return_type read() override { return return_type::OK; }
+  return_type read(std::vector<hardware_interface::Variant> & states) override
+  {
```

Comment:
Per my comment above, you might consider adding this:

  if (!states.is_initialized())
    return return_type::ERROR;

Reply:
I also don't like the return parameter being added to read here. It would be much better if `read` returned an **either** type (error or valid values).